core/src/main/java/com/amplifyframework/core/model/query/Where.java

```diff
@@ -63,6 +69,38 @@ public static QueryOptions id(@NonNull final String modelId) {
                 .paginated(Page.firstResult());
     }
 
+    /**
+     * Factory method that builds the options with a predicate matching the primary key and the
+     * pagination set to the first result only.
+     *
+     * @param itemClass model class.
+     * @param modelPrimaryKey model identifier.
+     * @param <T> Extends Model.
+     * @return options with proper predicate and pagination to match a model by its id.
+     * @throws AmplifyException Throws AmplifyException.
+     */
+    public static <T extends Model> QueryOptions identifier(@NonNull Class<T> itemClass,
+                                                           @NonNull final Serializable modelPrimaryKey)
+            throws AmplifyException {
+        final ModelSchema schema = ModelSchema.fromModelClass(itemClass);
+        final List<String> primaryKeyList = schema.getPrimaryIndexFields();
+        QueryOptions queryOptions = null;
+        Iterator<String> pkField = primaryKeyList.listIterator();
+        final QueryField idField = QueryField.field(pkField.next());
+        if (primaryKeyList.size() == 1 && !(modelPrimaryKey instanceof ModelPrimaryKey)) {
+            queryOptions = matches(idField.eq(Objects.requireNonNull((String) modelPrimaryKey)));
+        } else {
+            ModelPrimaryKey<?> primaryKey = (ModelPrimaryKey<?>) modelPrimaryKey;
+            Iterator<?> sortKeyIterator = primaryKey.sortedKeys().listIterator();
+            if (queryOptions == null) {
```

Comment:
Won't `queryOptions` always be null at this point?

Reply:
Resolved.